### PR TITLE
DEV: change ASV benchmarks to run on Python 3.6 by default

### LINKED ISSUE #11458: Use python 3 for benchmarks?
Currently, benchmarks are only done on python2 [1]. Shouldn't this be python3? Or at least not only python2?

[1] `asv.conf.json` has `"pythons": ["2.7"],`; see
https://github.com/numpy/numpy/blob/master/benchmarks/asv.conf.json#L38

### PR DESCRIPTION
**Pull request by shoyer**
Fixes #11458

### CODE REVIEW

**Comment by charris** [comment]
@pv Does this affect your tracking?

**Comment by tylerjereddy** [comment]
I could be wrong, but I think those benchmarks don't actually use the default values provided in the repo.

**Comment by charris** [comment]
Do we want this/need this variable? The documentation is a bit unclear, but it seems that asv will use whatever python is running it by default: https://asv.readthedocs.io/en/stable/asv.conf.json.html#pythons  I also think 3.7 may be rather new, most distros are probably supplying 3.6.

**Comment by tylerjereddy** [comment]
Sounds right, but if you provide a Python version in the json file it should use the one you specify irrespective of what you have active at the moment.

I guess it is good to have a sensible default -- 3.6 may be a little safer than 3.7 for the time being I suppose.

File: benchmarks/asv.conf.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
     // The Pythons you'd like to test against.  If not provided, defaults
     // to the current version of Python used to run `asv`.
-    "pythons": ["2.7"],
+    "pythons": ["3.7"],
```
Comment (charris) [inline benchmarks/asv.conf.json]:
```suggestion
    "pythons": ["3.6"],
```

**Comment by charris** [comment]
Thanks Stephan.

**Pull request merged**